bazaarvoice/emodb
# Feature: Hidden (aka Private) attributes

## Github Issue #

[91](https://github.com/bazaarvoice/emodb/issues/91)

## What Are We Doing Here?

I implemented the feature using the namespace strategy.

## How to Test and Verify 

1. Check out the project; run Emo
2. Write some documents with hidden attributes. 
3. Verify that you cannot see the hidden fields unless you specify showHiddenFields=true
4. Verify you cannot specify showHiddenFields=true unless you have update permission to the table
5. Verify you cannot see hidden fields in databus events or in stash
3. Write some conditional deltas, conditioned on the values of hidden attributes.
4. Verify Emo correctly computes the conditions (hidenness has no impact on conditional logic).

## Risk

### Level 

`Medium`

### Required Testing

`Smoke`, `Regression`, or `Manual`. (All changes except documentation need smoke
testing at a minimum).

Not sure????

I wrote a bunch of unit tests, but I couldn't figure out how to test stash, other than maybe to use test.integration.sor.ScanUploadTest. But the docs are apparently wrong. it requires 3 args instead of 2, and I have no clue what a `ddlFilePath` might be.

### Risk Summary

We are introducing a new concept to Emo. This is something we should do very, very carefully. Try to erase my use case from your mind and think about what a new user will make of this feature. Does the documentation make the intended usage obvious? Is the feature sufficiently intuitive?

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
